e2e/tests-dfx/deploy.bash

```diff
@@ -92,4 +92,11 @@ teardown() {
     assert_command_fail dfx deploy --specified-id n5n4y-3aaaa-aaaaa-p777q-cai
     assert_match "error: The following required arguments were not provided:"
     assert_match "<CANISTER_NAME>"
-} 
\ No newline at end of file
+}
+
+@test "deploy does not require wallet if all canisters are created" {
+    dfx_start
+    dfx canister create --all --no-wallet
+    assert_command dfx deploy
+    assert_not_match "Creating a wallet canister"
```

Comment:
This is a brittle verification: It would be easy to change the `Creating a wallet canister` log line, without updating this test, and then this test wouldn't verify anything.

```suggestion
    assert_not_contains "Creating a wallet canister"
    assert_command dfx identity get-wallet
    assert_contains "Creating a wallet canister"
```